sdk/data/azcosmos/cosmos_example_test.go

```diff
@@ -0,0 +1,136 @@
+
+	// Create new Cosmos DB client.
+	cred, _ := NewSharedKeyCredential(key)
+	client, err := NewCosmosClient(endpoint, cred, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// All operations for Go operate on a context.Context, allowing you to control cancellation/timeout.
+	ctx := context.Background()
+
+	// ===== 1. Creating a database =====
+
+	databaseName := CosmosDatabaseProperties{Id: "databaseName"}
+	database, err := client.CreateDatabase(ctx, databaseName, nil, nil)
+	log.Panic(err)
+
+	// This example showcases several common operations to help you get started, such as:
+	// ===== 2. Creating a container =====
+
+	properties := CosmosContainerProperties{
+		Id: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/id"},
+		},
+		IndexingPolicy: &IndexingPolicy{
+			IncludedPaths: []IncludedPath{
+				{Path: "/*"},
+			},
+			ExcludedPaths: []ExcludedPath{
+				{Path: "/\"_etag\"/?"},
+			},
+			Automatic:    true,
+			IndexingMode: IndexingModeConsistent,
+		},
+	}
+
+	throughput := NewManualThroughputProperties(400)
+
+	resp, _ := database.DatabaseProperties.Database.CreateContainer(ctx, properties, throughput, nil)
+	container := resp.ContainerProperties.Container
+
+	// ===== 2. Update container properties =====
+
+	resp, _ = container.Read(ctx, nil)
+
+	updatedProperties := CosmosContainerProperties{
+		Id: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/id"},
+		},
+		ETag: "someEtag",
+		IndexingPolicy: &IndexingPolicy{
+			IncludedPaths: []IncludedPath{},
+			ExcludedPaths: []ExcludedPath{},
+			Automatic:     false,
+			IndexingMode:  IndexingModeNone,
+		},
+	}
+
+	// Replace container properties
+	resp, _ = container.Replace(ctx, updatedProperties, nil)
+
+	// Read the manual throughput property
+	throughputResponse, _ := container.ReadThroughput(ctx, nil)
+	_, _ = throughputResponse.ThroughputProperties.ManualThroughput()
+
+	// Replace manual throughput property
+	newScale := NewManualThroughputProperties(500)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// Container autoscale throughput
+	_, _ = throughputResponse.ThroughputProperties.AutoscaleMaxThroughput()
+
+	newScale = NewAutoscaleThroughputProperties(10000)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// Delete container
+	resp, _ = container.Delete(ctx, nil)
+
+	// ===== 3. Item CRUD =====
+
+	// Items in an Azure Cosmos container are uniquely identified by their id and partition key value.
+	pk, _ := NewPartitionKey("1")
+
+	item := map[string]string{
+		"id":    "1",
+		"value": "2",
+	}
+
+	// Create item.
+	itemResponse, _ := container.CreateItem(ctx, pk, item, nil)
+
+	// Read item.
+	itemResponse, _ = container.ReadItem(ctx, pk, "1", nil)
+
+	// Item response body
+	var itemResponseBody map[string]interface{}
+	err = json.Unmarshal(itemResponse.Value, &itemResponseBody)
+	log.Panic(err)
+
+	// ===== 4. Optimistic Concurrency =====
```

Comment:
This is not Optimistic Concurrency, what you are doing in this section is Session consistency guarantee